src/applications/mhv/medical-records/containers/ConditionDetails.jsx

```diff
@@ -34,6 +40,27 @@ const ConditionDetails = props => {
   const { conditionId } = useParams();
   const dispatch = useDispatch();
 
+  const alertList = useSelector(state => state.mr.alerts?.alertList);
+  const [activeAlert, setActiveAlert] = useState();
+
+  useEffect(
+    () => {
+      if (alertList?.length) {
+        const filteredSortedAlerts = alertList
+          .filter(alert => alert.isActive)
+          .sort((a, b) => {
+            // Sort chronologically descending.
+            return b.datestamp - a.datestamp;
+          });
+        if (filteredSortedAlerts.length > 0) {
+          // The activeAlert is the most recent alert marked as active.
+          setActiveAlert(filteredSortedAlerts[0]);
+        }
+      }
+    },
+    [alertList],
+  );
```

Comment:
We have this exact same block of code for each of the domains where we are handling errors. Is it possible to factor the guts of this `useEffect` out into a helper function? You could pass it the `alertList` and a callback to `setActiveAlert`. If this is possible, please also implement it for Allergies and Vaccines.

Reply:
Great idea! I created a hook called "use-alerts" and implemented it wherever that useEffect was being used.

Reply:
Wow, nice! That's even a step beyond what I was thinking.